cg/apps/tb/api.py

```diff
@@ -1,16 +1,153 @@
+                "query": query,
+                "deleted": deleted,
+                "temp": temp,
+                "before": str(before) if before else None,
+                "is_visible": is_visible,
+                "family": family,
+                "data_analysis": data_analysis,
+            }
+        }
+        response = self.query_trailblazer(command="query-analyses", request_body=request_body)
+        if response:
+            if isinstance(response, list):
+                return [TrailblazerAnalysis.parse_obj(analysis) for analysis in response]
+            if isinstance(response, dict):
+                return [TrailblazerAnalysis.parse_obj(response)]
+        return response
+
+    def get_latest_analysis(self, case_id: str) -> Optional[TrailblazerAnalysis]:
+        request_body = {
+            "case_id": case_id,
+        }
+        response = self.query_trailblazer(command="get-latest-analysis", request_body=request_body)
+        if response:
+            return TrailblazerAnalysis.parse_obj(response)
+
+    def find_analysis(
+        self, case_id: str, started_at: dt.datetime, status: str
+    ) -> Optional[TrailblazerAnalysis]:
+        request_body = {"case_id": case_id, "started_at": str(started_at), "status": status}
+        response = self.query_trailblazer(command="find-analysis", request_body=request_body)
+        if response:
+            return TrailblazerAnalysis.parse_obj(response)
+
+    def get_latest_analysis_status(self, case_id: str) -> Optional[str]:
+        latest_analysis = self.get_latest_analysis(case_id=case_id)
+        if latest_analysis:
+            return latest_analysis.status
+
+    def has_latest_analysis_started(self, case_id: str) -> bool:
+        return self.get_latest_analysis_status(case_id=case_id) in self.__STARTED_STATUSES
+
+    def is_latest_analysis_ongoing(self, case_id: str) -> bool:
+        return self.get_latest_analysis_status(case_id=case_id) in self.__ONGOING_STATUSES
+
+    def delete_analysis(self, analysis_id: str, force: bool = False) -> None:
+        """Raises TrailblazerAPIHTTPError"""
+        request_body = {"analysis_id": analysis_id, "force": force}
+        self.query_trailblazer(command="delete-analysis", request_body=request_body)
+
+    def mark_analyses_deleted(self, case_id: str) -> Optional[list]:
+        """Mark all analyses for case deleted without removing analysis files"""
+        request_body = {
+            "case_id": case_id,
+        }
+        response = self.query_trailblazer(
+            command="mark-analyses-deleted", request_body=request_body
         )
+        if response:
+            if isinstance(response, list):
+                return [TrailblazerAnalysis.parse_obj(analysis) for analysis in response]
+            if isinstance(response, dict):
+                return [TrailblazerAnalysis.parse_obj(response)]
+
+    def add_pending_analysis(
+        self,
+        case_id: str,
+        type: str,
```

Comment:
Could/should the 'type' argument have a different name since it shadows a built in one?